src/Listener/ViewListener.php

```diff
@@ -34,7 +34,9 @@ public function initialize()
      */
     public function beforeFind(Event $event)
     {
-        $event->subject->query->contain($this->_getRelatedModels());
+        if (!$event->subject->query->contain()) {
+            $event->subject->query->contain($this->_getRelatedModels());
```

Comment:
Do you guys normally list `hasMany` associations on edit/view pages? I personally don't. If others also don't it would suggest changing this to `$this->_getRelatedModels(['manyToOne', 'oneToOne', 'belongsToMany'])`

@lorenzo @jadb @savant